src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/QueryParameterValueResolver.php

```diff
@@ -31,8 +31,12 @@ public function resolve(Request $request, ArgumentMetadata $argument): array
 
         $name = $attribute->name ?? $argument->getName();
         if (!$request->query->has($name)) {
-            if ($argument->isNullable() || $argument->hasDefaultValue()) {
-                return [];
+            if ($argument->hasDefaultValue()) {
+                return [$argument->getDefaultValue()];
```

Comment:
Can't we remove this "if" and let PHP do that part?

Reply:
I don't think this will work fine if that argument is not the last one of the controller signature.

Reply:
You mean to account for optional parameter declared before required parameters. This is deprecated by OK.

Reply:
Wouldn’t `ArgumentResolver` throw if we don’t return the default value here anyways?

Reply:
No, I mean  to account for optional parameters declared before another optional parameter for which an argument resolver would provide a value.